### PR TITLE
Rename a few APIs in FileSystem.Handle module

### PR DESCRIPTION
**Pull request by rnjtranjan**

### CODE REVIEW

**Comment by rnjtranjan** [comment]
Fixes: #1168

File: src/Streamly/Internal/FileSystem/Handle.hs [hunk 1]
```diff
@@ -35,16 +35,16 @@ module Streamly.Internal.FileSystem.Handle
...
+    , getBytesWith
 
     -- * Chunked Stream Read
     , readChunks
-    , readChunksWithBufferOf
+    , readChunksWith
 
-    , toChunksWithBufferOf
+    , toChunksWith
     , toChunks
```
Comment (harendra-kumar) [inline src/Streamly/Internal/FileSystem/Handle.hs:48]: Should be `getChunks` and `getChunksWith`.

**Review by harendra-kumar — CHANGES_REQUESTED** [review]

**Review by harendra-kumar — CHANGES_REQUESTED** [review]
Deprecate the released functions instead of changing them.

**Review by harendra-kumar — CHANGES_REQUESTED** [review]
Only those functions that have been released, should be deprecated. See https://streamly.composewell.com/haddocks/streamly-0.8.2/Streamly-FileSystem-Handle.html for released functions.

File: src/Streamly/FileSystem/Handle.hs [hunk 2]
```diff
@@ -76,9 +76,9 @@ module Streamly.FileSystem.Handle
     -- encountered.
 
     , read
-    , readWithBufferOf
+    , readWith
```
Comment (harendra-kumar) [inline src/Streamly/FileSystem/Handle.hs:79]: `readBufferOf` should be kept with deprecation. And same for other released functions as well.

**Review by harendra-kumar — CHANGES_REQUESTED** [review]

**Pull request merged**